syntax = "proto3";

option swift_prefix = "";
option java_package = "systems.ajax.a911.grpc.auth.v1";
option java_multiple_files = true;

import "v1/company/employee.proto";
import "v1/user/user.proto";
import "v1/common/traffic-logging.proto";
import "systems/ajax/logging/proto/log_marker.proto";
import "systems/ajax/logging/proto/formatting_options.proto";

message LoginRequest {
    string login = 1 [(hash_value) = true];
    string password = 2 [(hash_value) = true];

    // @since 1.1
    string machine_id = 3;
}

message CompanyLoginRequest {
    string company_id = 1 [(log_marker) = LOG_MARKER_COMPANY_ID];
    Employee.ComplexRole role = 2;

    // @since 1.1
    string machine_id = 3;
}

/*
 * Contains session information that should be used in all subsequent requests in session headers:
 *      x-session-user  - ID of the authenticated user. Required.
 *      x-session-token - session token. Required.
 *      x-user-agent    - name and version of the client. Required.
 */
message LoginResponse {
    string user_id = 1 [(log_marker) = LOG_MARKER_USER_ID];
    string session_token = 2 [(hash_value) = true];
    string company_id = 3 [(log_marker) = LOG_MARKER_COMPANY_ID];
    EmployeeInfo employee_info = 4;
    repeated FullUser.UserCompany user_companies = 5;
    FullUser.UserInfo user_info = 6;

    message EmployeeInfo {
        string employee_id = 1 [(log_marker) = LOG_MARKER_EMPLOYEE_ID];
        string first_name = 2 [(hash_value) = true];
        string last_name = 3 [(hash_value) = true];
        Employee.ComplexRole role = 4;
        string photo_id = 5;
        string cluster_company_id = 6 [(log_marker) = LOG_MARKER_COMPANY_ID];
    }
}
